core_lib/src/managers/selectionmanager.cpp

```diff
@@ -0,0 +1,370 @@
+        myTempTransformedSelection =
+            transformedSelection; // @ necessary?
+        QPointF anchorPoint = transformedSelection.center();
+        myRotatedAngle = ( atan2( currentPoint.y() - anchorPoint.y(), currentPoint.x() - anchorPoint.x() ) ) * 180.0 / M_PI - rotationOffset;
+        qDebug() << currentPoint << anchorPoint << myRotatedAngle << rotationOffset;
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+
+void SelectionManager::setSelection(QRectF rect)
+{
+    resetSelectionTransformProperties();
+    mySelection = rect;
+    myTransformedSelection = rect;
+    myTempTransformedSelection = rect;
+    mSomethingSelected = (mySelection.isNull() ? false : true);
+
+    emit selectionChanged();
+}
+
+void SelectionManager::calculateSelectionTransformation()
+{
+    QVector<QPoint> centerPoints = calcSelectionCenterPoints();
+
+    mSelectionTransform.reset();
+
+    mSelectionTransform.translate(centerPoints[0].x(), centerPoints[0].y());
+    mSelectionTransform.rotate(myRotatedAngle);
+
+    if (mySelection.width() > 0 && mySelection.height() > 0) // can't divide by 0
+    {
+        float scaleX = myTempTransformedSelection.width() / mySelection.width();
+        float scaleY = myTempTransformedSelection.height() / mySelection.height();
+        mSelectionTransform.scale(scaleX, scaleY);
+    }
+    mSelectionTransform.translate(-centerPoints[1].x(), -centerPoints[1].y());
+}
+
+QVector<QPoint> SelectionManager::calcSelectionCenterPoints()
+{
+    QVector<QPoint> centerPoints;
+    float selectionCenterX,
+        selectionCenterY,
+        tempSelectionCenterX,
+        tempSelectionCenterY;
+
+    tempSelectionCenterX = myTempTransformedSelection.center().x();
+    tempSelectionCenterY = myTempTransformedSelection.center().y();
+    selectionCenterX = mySelection.center().x();
+    selectionCenterY = mySelection.center().y();
+    centerPoints.append(QPoint(tempSelectionCenterX, tempSelectionCenterY));
+    centerPoints.append(QPoint(selectionCenterX, selectionCenterY));
+    return centerPoints;
+}
+
+
+QPointF SelectionManager::offsetFromAspectRatio(qreal offsetX, qreal offsetY)
+{
+    qreal factor = myTransformedSelection.width() / myTransformedSelection.height();
+
+    if (mMoveMode == MoveMode::TOPLEFT || mMoveMode == MoveMode::BOTTOMRIGHT)
+    {
+        offsetY = offsetX / factor;
+    }
+    else if (mMoveMode == MoveMode::TOPRIGHT || mMoveMode == MoveMode::BOTTOMLEFT)
+    {
+        offsetY = -(offsetX / factor);
+    }
+    else if (mMoveMode == MoveMode::MIDDLE)
+    {
+        qreal absX = offsetX;
+        if (absX < 0) { absX = -absX; }
+
+        qreal absY = offsetY;
+        if (absY < 0) { absY = -absY; }
+
+        if (absX > absY) { offsetY = 0; }
+        if (absY > absX) { offsetX = 0; }
+    }
+    return QPointF(offsetX, offsetY);
+}
+
+/**
+ * @brief ScribbleArea::flipSelection
+ * flip selection along the X or Y axis
+*/
+void SelectionManager::flipSelection(bool flipVertical)
```

Comment:
Not important, but it might be nice to use `Qt:Orientation` or `Qt:Orientations` here instead of a boolean.